lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2568,6 +2568,97 @@ class ConvertAtenPermuteOp : public OpConversionPattern<AtenPermuteOp> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenSliceTensorOp : public OpConversionPattern<AtenSliceTensorOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenSliceTensorOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    Location loc = op.getLoc();
+    TypeConverter *typeConverter = getTypeConverter();
+
+    auto input = adaptor.self();
+    RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+    RankedTensorType resultType =
+        typeConverter->convertType(op->getResult(0).getType())
+            .cast<RankedTensorType>();
+    int64_t resultRank = resultType.getRank();
+    SmallVector<Value> resultShape;
+    Value zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    Value one = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+
+    SmallVector<Value> offsets, strides;
+    strides.resize(inputType.getRank(), one);
+    offsets.resize(inputType.getRank(), zero);
+    Value end, start, stepIndex;
```

Comment:
The code to figure out `end`, `start`, `stepIndex` are quite several lines and independant to reach other.  I would put `Value end` followed by code to get the `end`, `Value start` followed by code to get the `start` and so on. Also each section should be separated by spaces. This would improve the readability.